common/flockfile/src/lib.rs

```diff
@@ -0,0 +1,116 @@
+use log::{debug, warn};
+use nix::fcntl::{flock, FlockArg};
+use std::{
+    fs::{self, File, OpenOptions},
+    io,
+    os::unix::io::AsRawFd,
+    path::{Path, PathBuf},
+};
+
+#[derive(thiserror::Error, Debug)]
+pub enum FlockfileError {
+    #[error(transparent)]
+    IoError(#[from] std::io::Error),
+
+    #[error("Couldn't acquire file lock.")]
+    NixError(#[from] nix::Error),
+}
+
+/// flockfile creates a lockfile in the filesystem under `/run/lock` and then creates a filelock using system fcntl with flock.
+/// flockfile will automatically remove lockfile on application exit and the OS should cleanup the filelock afterwards.
+/// If application exits unexpectedly the filelock will be dropped, but the lockfile will not be removed unless handled in signal handler.
+#[derive(Debug)]
+pub struct Flockfile {
+    handle: Option<File>,
+    pub path: PathBuf,
+}
+
+impl Flockfile {
+    /// Create new lockfile in `/run/lock` with specific name:
+    ///
+    /// #Example
+    ///
+    /// let _lockfile = match flockfile::Flockfile::new_lock("app")).unwrap();
+    ///
+    pub fn new_lock(lock_name: impl AsRef<Path>) -> Result<Flockfile, FlockfileError> {
+        let path = Path::new("/run/lock").join(lock_name);
```

Comment:
This is great question, since Linux Filesystem Hierarchy Standard 3.0 which has been released in 2015 they should if they adhere to it. Some may still use outdated `/var/lock` though, Debian uses `/run` hierarchy and so sholud its derivatives (Raspberrypi OS does as well as Ubuntu).
I used `/run` as systemd uses `/run` for its `pidfiles` as well and I believe it should be better to keep the files in fairly concise locations.